core/src/main/scala/org/apache/spark/ExecutorAllocationManager.scala

```diff
@@ -696,7 +717,8 @@ private[spark] class ExecutorAllocationManager(
 
```

Comment:
I think we should remove the stage attempt from unschedulableTaskSets when the stage completes in case it fails or something and never send the unschedulabletasksetremoved

Reply:
I think as long as the stageattempt is schedulable we can remove it, although it can fail but eventually if it becomes unschedulable again, we would add it back again to unschedulableTaskSets right? Wouldn't that be simpler?